packages/core/src/components/segmented-control/segmentedControl.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import classNames from "classnames";
+import * as React from "react";
+
+import { Classes, Intent } from "../../common";
+import { DISPLAYNAME_PREFIX, type HTMLDivProps, type Props, removeNonHTMLProps } from "../../common/props";
+import { Button } from "../button/buttons";
+
+/**
+ * SegmentedControl component props.
+ */
+export interface SegmentedControlProps<T extends string>
+    extends Props,
+        HTMLDivProps,
+        React.RefAttributes<HTMLDivElement> {
+    options: Array<{
+        id: T;
+        label: string;
+    }>;
```

Comment:
I think this should use `OptionProps` like RadioGroup and HTMLSelect. I will push a commit to change this.